pkg/apis/core/validation/shoot_test.go

```diff
@@ -487,6 +487,95 @@ var _ = Describe("Shoot Validation Tests", func() {
 			})
 		})
 
+		Context("#ValidateForceDeletion", func() {
+			It("should not allow setting the force-deletion annotation if the Shoot does not have a deletionTimestamp", func() {
+				newShoot := prepareShootForUpdate(shoot)
+
+				metav1.SetMetaDataAnnotation(&newShoot.ObjectMeta, v1beta1constants.AnnotationConfirmationForceDeletion, "1")
+
+				err := ValidateForceDeletion(newShoot, shoot)
+				Expect(err).To(PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":   Equal(field.ErrorTypeForbidden),
+					"Field":  Equal("metadata.annotations[confirmation.gardener.cloud/force-deletion]"),
+					"Detail": Equal("force-deletion annotation cannot be set when Shoot deletionTimestamp is nil"),
+				})))
+			})
+
+			It("should not allow setting the force-deletion annotation if the Shoot status does not have an ErrorCode", func() {
```

Comment:
```suggestion
			It("should not allow setting the force-deletion annotation if the Shoot status does not have an error code", func() {
```

also fix other occurrences

Why do you always write it like this? 😄